### PR TITLE
Refactor utils

### PR DESCRIPTION
**Pull request by chanhosuh**
### Description

Carved out spaces to house common utilities and constants/enums.  This is to help ensure standardized usage of common python and web3 libraries.

### Hygiene checklist

- [x] Changelog entry
- [x] Everything public has a Numpy-style docstring
      (modules, public functions, classes, and public methods)
- [x] Commit history is cleaned-up with minor changes squashed together
      and descriptive commit messages following [Tim Pope's style](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html)

### Cute Animal Picture

Put a link to a cute animal picture inside the parenthesis-->

### CODE REVIEW

File: changelog.d/20231031_152630_chanhosuh_refactor_utils.rst [hunk 1]
```diff
@@ -0,0 +1,5 @@
+Added
+-----
+- Created utilities for managing datetime, addresses, and enums.
+- Created a constants module to house common constants / enums.
```
Comment (allt0ld) [inline changelog.d/20231031_152630_chanhosuh_refactor_utils.rst:4]: Would be helpful to name which enums

File: curvesim/pool_data/__init__.py [hunk 2]
```diff
@@ -45,28 +49,34 @@ def get_data_cache(address, chain="mainnet", days=60, end=None):
...
-        Pool address prefixed with “0x”.
+    address : str, Address
+        Pool address in proper checksum hexadecimal format.
 
-    chain : str
+    chain : str, Chain
         Chain/layer2 identifier, e.g. “mainnet”, “arbitrum”, “optimism".
 
+    end_ts : int
+        Posix timestamp
```
Comment (allt0ld) [inline curvesim/pool_data/__init__.py:69]: Could write what the timestamp means

File: curvesim/constants.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+    def __str__(self):
+        """
+        Regular Enum's __str__ is the name, rather than the value,
+        e.g.
+
+        >>> str(Chain.MAINNET)
+        'Chain.MAINNET'
+
+        so we need to explicit use the value.
```
Comment (allt0ld) [inline curvesim/constants.py:21]: so we need to explicitly use the value

File: curvesim/utils/__init__.py [hunk 4]
```diff
@@ -1,16 +1,29 @@
 """Utlity functions for general usage in Curvesim."""
+__all__ = [
+    "get_env_var",
+    "get_pairs",
+    "dataclass",
+    "get_event_loop",
+    "cache",
+    "override",
+    "datetime",
```
Comment (allt0ld) [inline curvesim/utils/__init__.py:9]: Perhaps I'm missing something, but it seems datetime isn't imported

File: curvesim/constants.py (hunk 3 above)
Comment (allt0ld) [inline curvesim/constants.py:21]: "so we need to explicitly use the value"

File: curvesim/pool_data/__init__.py (hunk 2 above)
Comment (allt0ld) [inline curvesim/pool_data/__init__.py:69]: Could write what the timestamp is supposed to mean

File: curvesim/utils/__init__.py (hunk 4 above)
Comment (allt0ld) [inline curvesim/utils/__init__.py:9]: Perhaps I'm missing something, but it seems `.datetime` isn't imported below

File: curvesim/pool_data/__init__.py [hunk 5]
```diff
@@ -45,28 +49,34 @@ def get_data_cache(address, chain="mainnet", days=60, end=None):
 
 
 def get_metadata(
```
Comment (allt0ld) [inline curvesim/pool_data/__init__.py:51]: There is a merge conflict as #277  moved `get_metadata` to a new file `curvesim.pool_data.queries.metadata` and then imported it into this file from there

**Review by allt0ld — APPROVED** [review]

File: curvesim/utils/__init__.py (hunk 4 above)
Comment (chanhosuh) [reply curvesim/utils/__init__.py:9]: It's fine because it becomes part of the namespace due to how modules are loaded.

**Pull request merged**